buildbot/master/files/config/factories.py

```diff
@@ -103,6 +103,86 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps
+    """
+    yaml_path = ""
+
+    def __init__(self, builder_name, environment, yaml_path):
+        print(yaml_path)
+        self.builder_name = builder_name
+        self.environment = environment
+        self.yaml_path = yaml_path
+
+    def run(self):
+        print(self.yaml_path)
+        try:
+            full_yaml_path = os.path.join(self.build.workdir, self.yaml_path)
+            with open(full_yaml_path) as steps_file:
+                builder_steps = yaml.safe_load(steps_file)
+            commands = builder_steps[self.builder_name]
+            dynamic_steps = [self.make_step(command) for command in commands]
+        except Exception as e:  # Bad step configuration, fail build
+            print(str(e))
+            dynamic_steps = [BadConfigurationStep(e)]
```

Comment:
Instead of creating a `BadConfigurationStep` to raise an exception, we should just raise the exception here since we're already in a step. (It was a separate step before because the parsing was done at Buildbot start-up time, and raising an exception them would kill the Buildbot master.)